Add feature to select which subset of config is cloned/merged/deployed

pizen opened the pull request:
Adds --include and --exclude CLI parameters to specify which categories
of config data should be part of the operation. These parameters are
mutually exclusive and take CSV lists of category names, which are
documented in the help strings. When using --include only those
categories are used. With --exclude it's all but what's specified.

Update unit tests for new functionality.
Update README.md with new help strings.

Fixes #5

pizen committed:
Add feature to select which subset of config is cloned/merged/deployed

Adds --include and --exclude CLI parameters to specify which categories
of config data should be part of the operation. These parameters are
mutually exclusive and take CSV lists of category names, which are
documented in the help strings. When using --include only those
categories are used. With --exclude it's all but what's specified.

Update unit tests for new functionality.
Update README.md with new help strings.

Fixes #5

tgayvert reviewed (commented):
This looks good and works as I expect it to. Does the `--include` and `--exclude` play nicely with the wildcard options from the other PR?

pizen commented:
> Does the `--include` and `--exclude` play nicely with the wildcard options from the other PR?

It should but I'll test before merging.

pizen committed:
Merge branch 'master' into feature/selective_operation

pizen committed:
Fix issue with backup/merge when target has no advanced rules

Discovered during testing of #15

pizen committed:
Fix issue from incorrectly resolving merge conflict

pizen committed:
Indicate when advanced rules are skipped

Makes it easier to tell when advanced rules are skipped in a merge.
Before if there was no advanced rules to merge it appeared as if this
entire step was skipped because there was no output at all.

pizen commented:
@tgayvert Updated
